src/main/java/com/artipie/RepositoriesFromStorage.java

```diff
@@ -4,19 +4,65 @@
  */
 package com.artipie;
 
+import com.artipie.asto.Content;
 import com.artipie.asto.Key;
 import com.artipie.asto.Storage;
+import com.artipie.asto.ext.PublisherAs;
 import com.artipie.repo.ConfigFile;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
 import hu.akarnokd.rxjava2.interop.SingleInterop;
 import io.reactivex.Single;
+import java.util.Objects;
 import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Artipie repositories created from {@link Settings}.
  *
  * @since 0.13
  */
 public final class RepositoriesFromStorage implements Repositories {
+    /**
+     * Cache for config files.
+     */
+    private static LoadingCache<KeyAndStorage, Single<String>> configs;
+
+    /**
+     * Cache for aliases.
+     */
+    private static LoadingCache<KeyAndStorage, Single<StorageAliases>> aliases;
+
+    static {
+        final CacheLoader<KeyAndStorage, Single<String>> ldrconfigs = new CacheLoader<>() {
+            @Override
+            public Single<String> load(final KeyAndStorage pair) {
+                return Single.fromFuture(
+                    new ConfigFile(pair.key)
+                        .valueFrom(pair.storage)
```

Comment:
I don't like private fields access here because it increases the coupling between components. I'd suggest introducing a method in `KeyAndStorage` which can access these private fields, e.g.
```
// class KeyAndStorage
CompletionStage<Content> content() {
  return new ConfigFile(this.key).valueFrom(this.storage);
}
```
Also, in this case, it could have a more meaningful name, e.g. `ConfigContent` or similar